src/qtsoap.h

```diff
@@ -309,8 +227,8 @@ class QT_QTSOAP_EXPORT QtSoapArrayIterator
     int pos() const;
     void pos(int *pos0, int *pos1 = 0, int *pos2 = 0, int *pos3 = 0, int *pos4 = 0) const;
 
-    QtSoapType *data();
-    const QtSoapType *current() const;
+    std::shared_ptr<QtSoapType> data();
```

Comment:
Where they jumping the bridge here and returning the raw point directly? O__o
Ah well, I think it's better like this.